System/Console/ANSI/Types.hs

```diff
@@ -37,28 +38,26 @@ data ColorIntensity = Dull
                     | Vivid
                     deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
-
+-- | ANSI allows for a palette of up to 256 colors
 newtype Color8Code = Color8Code Word8 deriving (Eq, Show, Read)
 
--- | Represents 8-bit ANSI colors whose codes are in the range 0x10 - 0xFF.
+-- | Represents the xterm 256 color palette protocol.
 --
---  For reference, here are all ranges of 8-bit ANSI colors as defined in
+--  For reference, here are all ranges of colors as defined in
 --  https://en.wikipedia.org/wiki/ANSI_escape_code#Colors lists the ranges:
 --
--- 0x00-0x07:  standard colors (as in ESC [ 30–37 m)
--- 0x08-0x0F:  high intensity colors (as in ESC [ 90–97 m)
--- 0x10-0xE7:  6 × 6 × 6 cube (216 colors): 16 + 36 × r + 6 × g + b (0 ≤ r, g, b ≤ 5)
--- 0xE8-0xFF:  grayscale from black to white in 24 steps
---
---  The 8-bit ANSI colors whose codes are in the 0x00-0x0F range are equivalent to the
---  ANSI 4-bit colors, hence they are not represented by Color8, but by
---  the pair (ColorIntensity, Color).
+-- 0x00-0x07:  standard (system) colors (as in ESC [ 30–37 m)
+-- 0x08-0x0F:  high intensity (system) colors (as in ESC [ 90–97 m)
+-- 0x10-0xE7:  6 × 6 × 6 cube (216 colors):
+--             16 + 36 × r + 6 × g + b (0 ≤ r, g, b ≤ 5)
+-- 0xE8-0xFF:  grayscale from dark gray to near white in 24 steps
 --
-data Color8 = RGB8Color !Word8 !Word8 !Word8 -- ^ ANSI 8-bit "6 × 6 × 6 cube" (216 colors) rgb.
-                                       --   r,g,b components are in range [0..5].
-            | Gray8Color !Word8          -- ^ ANSI 8-bit "grayscale" colors.
-                                       --   gray component is in range [0..23].
-            deriving (Eq, Show, Read)
+data Xterm256Color = SystemColor ColorIntensity Color
+                   | RGBColor (RGB Word8) -- ^ ANSI 8-bit "6 × 6 × 6 cube" (216 colors) rgb.
```

Comment:
This one also could be strict

Reply:
The type `RGB a` in the `colour` pacakge is implemented with strict fields. See https://hackage.haskell.org/package/colour-2.3.3/docs/Data-Colour-SRGB.html#t:RGB.

Reply:
yes but we could put a strictness annotation here too? Or does it "propagate" from the underlying type?